src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -278,10 +283,18 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens): void
                     $classyKinds[] = T_ENUM;
                 }
 
+                $openedCurlyBrackets = 0;
+                $this->reservedIdentifiers = [];
+
                 for ($index = $namespace->getScopeStartIndex(); $index < $namespace->getScopeEndIndex() + $indexDiff; ++$index) {
                     $origSize = \count($tokens);
 
-                    if ($discoverSymbolsPhase && $tokens[$index]->isGivenKind($classyKinds)) {
+                    if ($tokens[$index]->equals('{')) {
+                        ++$openedCurlyBrackets;
+                    } if ($tokens[$index]->equals('}')) {
+                        unset($this->reservedIdentifiers[$openedCurlyBrackets]);
+                        --$openedCurlyBrackets;
+                    } elseif ($discoverSymbolsPhase && $tokens[$index]->isGivenKind($classyKinds)) {
```

Comment:
If I understand correctly, each set of reserved identifiers exists only between `{` and `}`, these are effectively merged as long as current token index is inside `{}`, and last one is unset when closing brace is found. That means the set of identifiers grows with each nested `{}`.

My concerns about this (thinking out loud) is: should these identifiers always be merged from first `{` up to the last (current) `{`? I mean, test cases don't have `@template` on class level, and it's just my guess, but if there's a template `T` on class level and then somewhere inside you have anonymous class with template `T2`, the `T` symbol should be allowed to use within that anonymous class since it's a different scope 🤔? It's getting too complicated for token-based logic 😅.

Reply:
The implementation should be how it works in reality. Can you provide a https://phpstan.org/try repro?

Reply:
Don't have time to figure it out, so maybe we can just improve it _if_ problem occurs 😉.

Reply:
Thank you ❤